windows/devices/uwb/UwbSession.cxx

```diff
@@ -32,6 +174,57 @@ UwbSession::ConfigureImpl(const uwb::protocol::fira::UwbSessionData& uwbSessionD
     }
 
     m_sessionId = uwbSessionData.sessionId;
+
+    // Populate the PUWB_SET_APP_CONFIG_PARAMS
+    auto sessionUwbMap = uwbSessionData.uwbConfiguration.GetValueMap();
+
+    std::vector<std::shared_ptr<uint8_t>> datavector(ddi_service_map.size());
```

Comment:
I'd like to see this code move to a helper function in some form whose job is to take a `UwbSessionData` object and output a `UWB_SET_APP_CONFIG_PARAMS` structure:

```
UWB_SET_APP_CONFIG_PARAMS
GenerateUwbSetAppConfigParameterDdi(const UwbSessionData& uwbSessionData)
```

So far, we've kept this kind of code outside of class'es interfaces and implemented them in detail namespaces in the source files they are needed to separate their concerns (see [here](https://github.com/microsoft/nearobject-framework/blob/27606804546b25b20606f23a045615b2ffbd1232/windows/devices/uwb/UwbDevice.cxx#L42-L49) for an example. In this case, `UwbSessionData` itself shouldn't need to be aware that the UwbCx DDI exists at all.

We should take that one step further and implement them independently in a helper library of some sort (eg. `libuwbcxadaption` or something). This way, we can independently test that functionality and isolate changes to the DDI to just that library.